batch_jobs/dcf_replication/dcf_replication_job.py

```diff
@@ -0,0 +1,390 @@
+import os
+import sys
+
+import csv
+from functools import partial
+from multiprocessing.pool import Pool
+import time
+import logging
+import io
+import datetime
+
+import boto3
+from botocore.exceptions import ClientError
+
+from batch_jobs.bin.settings import (
+    POSTFIX_1_EXCEPTION,
+    POSTFIX_2_EXCEPTION,
+    PROJECT_ACL,
+    GDC_TOKEN,
+)
+
+logging.basicConfig(level=logging.INFO)
+# logging.getLogger().addHandler(logging.StreamHandler(sys.stdout))
+
+NUMBER_OF_THREADS = 16
+MAX_RETRIES = 10
+JOB_STATUS_KEY = "job_status"
+REGION = os.environ.get("REGION", "us-east-1")
+
+
+def run_job(
+    manifest_file,
+    job_queue,
+    job_definition,
+    output_manifest_bucket,
+    thread_count=NUMBER_OF_THREADS,
+    max_retries=MAX_RETRIES,
+):
+    """
+    Start to run an job to generate bucket manifest
+    Args:
+        file_info(dict): file info
+        job_queue(str): job queue name
+        job_definition(str): job definition name
+
+    Returns:
+        bool: True if the job was submitted successfully
+    """
+    global NUMBER_OF_THREADS
+    global MAX_RETRIES
+
+    NUMBER_OF_THREADS = int(thread_count)
+    MAX_RETRIES = int(max_retries)
+    logging.info(
+        f"Submission Job started at {datetime.datetime.now()} with {NUMBER_OF_THREADS} threads and {MAX_RETRIES} retries."
+    )
+    local_manifest = get_manifest_from_bucket(manifest_file)
+    parsed_data = parse_manifest_file(local_manifest)
+    submitted, skipped, failed = submit_jobs(
+        parsed_data, job_queue, job_definition, output_manifest_bucket
+    )
+
+    logging.info(f"Job submission summary:")
+    logging.info(f"Submitted: {submitted} jobs")
+    logging.info(f"Skipped: {skipped} files (already exist)")
+    logging.info(f"Failed: {failed} submissions")
+    logging.info(f"Job ending time is {datetime.datetime.now()}")
+
+
+def submit_job(job_queue, job_definition, file):
+    key = file["id"] + "/" + file["file_name"]
+    session = boto3.Session(profile_name="default")
+    s3 = session.client("s3")
+
+    # Pre-check if bucket exists
+    if not check_bucket_exists(s3, file["destination_bucket"]):
+        logging.error(
+            "Destination bucket does not exist in s3: {}".format(
+                file["destination_bucket"]
+            )
+        )
+        file[JOB_STATUS_KEY] = "SKIPPED"
+        return file
+
+    # Pre-check if file already exists
+    exists, message = check_file_exists(
+        s3,
+        file["destination_bucket"],
+        key,
+        int(file["size"]),
+        file["md5"],
+    )
+    print(f"file_key is {key}, result is {message}")
+
+    if exists:
+        logging.info(f"Skipping {key}: {message}")
+        file[JOB_STATUS_KEY] = "SKIPPED"
```

Comment:
so these two cases share the same status, I think it would be nice to change the previous

Reply:
Done